slackbot/dispatcher.py

```diff
@@ -138,9 +138,23 @@ def loop(self):
         while True:
             events = self._client.rtm_read()
             for event in events:
-                if event.get('type') != 'message':
+                try:
+                    if event['type'] == 'message':
+                        self._on_new_message(event)
+                    elif event['type'] in ['channel_created', 'group_joined', 'im_created']:
+                        channel = [event['channel']]
+                        self._client.parse_channel_data(channel)
+                    elif event['type'] == 'team_join':
+                        user = event['user']
+                        self._client.users[user['id']] = user
+                    else:
+                        continue
+                except KeyError:
```

Comment:
I see your note about not being a fan of get, but in this case, there's too much other code being called that could raise a keyerror.